test/kontrol/README.md

```diff
@@ -0,0 +1,200 @@
+This example instructs on how to offload the initial part of your proof computation to Foundry (which is blazing fast) and then incorporate it into a Kontrol proof.
+
+The way we achieve this is via the cheatcodes `vm.stopAndReturnStateDiff` or `vm.dumpState` (undocumented at the time of writing).
+Depending on the cheatcode used the approach is slightly different. However, in both cases we produce a JSON containing the state of the chain after executing some code, and load that state into Kontrol.
+In this example, we are using `vm.dumpState` since it's easier to work with. For complete instructions on how to include external computation with `vm.stopAndReturnStateDiff`, please see the [documentation](https://github.com/runtimeverification/kontrol/tree/master/docs/external-computation).
+
+We will be using external computation to prove that the `asset()` function of `EVault` never reverts, in accordance with the ERC4626 specification.
+Even though the property itself is simple, setting up the state of `EVault` and supplementary contracts is time- and resource-intensive. We will use Foundry to set up the state and then load it into Kontrol to prove the property.
+
+To use external computation in Kontrol, one needs to:
+- Add the following permissions to `foundry.toml`
+```toml
+fs_permissions = [{ access = "read-write", path = "./"}]
+```
+- Add `vm.dumpState` cheatcode to the function that performs deployment and state setup for the test contract. In this case, it's the `_runSetUp()` function in [`init-state/InitState.sol`](./init-state/InitState.sol).
+- Add calls to the `saveAddresses` function implemented in [`init-state/SaveAddresses.sol`](./init-state/SaveAddresses.sol) to save the addresses of the contracts we want to use in the proof. `saveAddresses` is called with two parameters: the address of the contract and the name of the contract, which can be used in the proof to reference its address.
+- Run `forge script test/kontrol/init-state/InitState.sol:InitState --sig runSetUp --ffi` to execute this function and generate JSON files ([`init-state/InitState.json`](./init-state/InitState.json) and [`init-state/AddressNames.json`](./init-state/AddressNames.json)) with state updates corresponding to the deployment process and saved addresses. 
+- Generate helper contracts, converting the generated JSON to a Solidity contract. This is done by running
+```shell
+kontrol load-state InitState test/kontrol/init-state/InitState.json --contract-names test/kontrol/init-state/AddressNames.json --output-dir test/kontrol/proofs/utils`.
+```
+This will generate two Solidity contracts in the [proofs/utils](./proofs/utils) directory that can be used to load the state into Kontrol.
+- You can then load the state into Kontrol by implementing a proof contract that inherits the [proofs/utils/InitState.sol](./proofs/utils/InitState.sol) contract generated in the previous step:
```

Comment:
I think this is incorrect since inheriting from `InitState` only gives access to the names of the addresses. To load the state, you have to either run the `recreateState` function or prove with the `--init-state-from-dump`. So maybe rephrase this as something like:
> Write a proof contract that inherits [proofs/utils/InitState.sol](./proofs/utils/InitState.sol), generated in the previous step, to access the names of the deployed contracts. Note that the generated contract has a `recreateState` function to recreate the state updates recorded with Foundry. However, it is faster to add said state updates through the `kontrol prove` command as instructed below.

Reply:
Done!